src/org/rust/lang/core/psi/impl/RustNamedElementImpl.kt

```diff
@@ -19,7 +19,7 @@ public abstract class RustNamedElementImpl(node: ASTNode)   : RustCompositeEleme
         throw UnsupportedOperationException();
     }
 
-    override fun getNavigationElement() = getNameElement()
+    override fun getNavigationElement() = getNameElement() ?: this
```

Comment:
This isn't strictly related to this PR, but without it `go to definiton` does not work for modules in separate files (because  `nameElement` is null)